src/tree/ContainerAppTreeItem.ts

```diff
@@ -78,11 +78,11 @@ export class ContainerAppTreeItem extends AzExtParentTreeItem implements IAzureR
 
     public async browse(): Promise<void> {
         // make sure that ingress is enabled
-        if (!this.data.latestRevisionFqdn) {
+        if (this.ingressEnabled()) {
             throw new Error(localize('enableIngress', 'Enable ingress to perform this action.'));
         }
 
-        await openUrl(`https://${this.data.latestRevisionFqdn}`);
+        await openUrl(`https://${this.data.configuration?.ingress?.fqdn}`);
```

Comment:
If any of this chain is `undefined`, it'll try to open `https://` which is probably going to have weird results. If we are confident that the parts can never be `undefined`, I'd use `!` to assert it; otherwise I'd guard this with an `if`.